### PR TITLE
[FIX] Filter for only `ImagingSession`s or `PhenotypicSession`s in SPARQL query

### LINKED ISSUE #361: Add submodule checkout step to CI test suite
When our CI suite runs on GH, we want to make use of the new SPARQL test graph introduced in #357. However, this new graph depends on two new submodules (also introduced in #357) to be available. We therefore have to

- [ ] Update the CI test workflow to initialize the new submodules
- [x] Launch the local test graph before the tests are run in the CI
- [x] Ensure all tests are run, including the integration tests (the ones that use a real graph instead of mocking graph responses)

_Originally posted by @alyssadai in https://github.com/neurobagel/api/pull/357#discussion_r1809958658_

### LINKED ISSUE #374: SPARQL query matches same session too many times due to subclass inferencing
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Expected Behavior

If a specific session matches my query and has both imaging and phenotypic data, I expect to see in the unaggregated API results two entries for that subject-session: one `ImagingSession` and one `PhenotypicSession` instance.

### Current Behavior

When the query is going to a graph that also has the [Neurobagel vocabulary](https://github.com/neurobagel/recipes/blob/main/vocab/nb_vocab.ttl) in it, a matching session (unexpectedly) has **3** instead of two entries in the results: one `ImagingSession`, one `PhenotypicSession`, and one `Session` instance.

Note that the number of matching subjects, as well as the `num_matching_{phenotypic,imaging}_sessions` for a particular subject are still calculated correctly. The main problem is that the extra session instance (which also appears as an extra row in a participant-level results TSV from the query tool) gives the impression of 3 different session types, which is not how the data is modeled in the graph. 

e.g.,
```json
    ...
    "subject_data": [
      {
        "sub_id": "sub-01",
        "session_id": "ses-01",
        "num_matching_phenotypic_sessions": 2,
        "num_matching_imaging_sessions": 2,
        "session_type": "http://neurobagel.org/vocab/ImagingSession",
        "age": null,
        "sex": null,
        "diagnosis": [
          null
        ],
        "subject_group": null,
        "assessment": [
          null
        ],
        "image_modal": [
          "http://purl.org/nidash/nidm#T1Weighted",
          "http://purl.org/nidash/nidm#FlowWeighted"
        ],
        "session_file_path": "/data/neurobagel/bagel-cli/bids-examples/synthetic/sub-01/ses-01",
        "completed_pipelines": {
          "https://github.com/nipoppy/pipeline-catalog/tree/main/processing/fmriprep": [
            "23.1.3"
          ],
          "https://github.com/nipoppy/pipeline-catalog/tree/main/processing/freesurfer": [
            "7.3.2"
          ]
        }
      },
      {
        "sub_id": "sub-01",
        "session_id": "ses-01",
        "num_matching_phenotypic_sessions": 2,
        "num_matching_imaging_sessions": 2,
        "session_type": "http://neurobagel.org/vocab/PhenotypicSession",
        "age": 34.1,
        "sex": "http://purl.bioontology.org/ontology/SNOMEDCT/248152002",
        "diagnosis": [
          null
        ],
        "subject_group": "http://ncicb.nci.nih.gov/xml/owl/EVS/Thesaurus.owl#C94342",
        "assessment": [
          "https://www.cognitiveatlas.org/task/id/trm_57964b8a66aed",
          "https://www.cognitiveatlas.org/task/id/tsk_4a57abb949ece"
        ],
        "image_modal": [
          null
        ],
        "session_file_path": null,
        "completed_pipelines": {}
      },
      {
        "sub_id": "sub-01",
        "session_id": "ses-01",
        "num_matching_phenotypic_sessions": 2,
        "num_matching_imaging_sessions": 2,
        "session_type": "http://neurobagel.org/vocab/Session",
        "age": 34.1,
        "sex": "http://purl.bioontology.org/ontology/SNOMEDCT/248152002",
        "diagnosis": [
          null
        ],
        "subject_group": "http://ncicb.nci.nih.gov/xml/owl/EVS/Thesaurus.owl#C94342",
        "assessment": [
          "https://www.cognitiveatlas.org/task/id/trm_57964b8a66aed",
          "https://www.cognitiveatlas.org/task/id/tsk_4a57abb949ece",
          null
        ],
        "image_modal": [
          null,
          "http://purl.org/nidash/nidm#T1Weighted",
          "http://purl.org/nidash/nidm#FlowWeighted"
        ],
        "session_file_path": "/data/neurobagel/bagel-cli/bids-examples/synthetic/sub-01/ses-01",
        "completed_pipelines": {
          "https://github.com/nipoppy/pipeline-catalog/tree/main/processing/fmriprep": [
            "23.1.3"
          ],
          "https://github.com/nipoppy/pipeline-catalog/tree/main/processing/freesurfer": [
            "7.3.2"
          ]
        }
      },
      ...
```

### Error message

_No response_

### Environment

- OS:
- Python/Node version:

### How to reproduce

_No response_

### Anything else?

This happens due to the new class relationships we have in the graph, specifically
```ttl
nb:ImagingSession a rdfs:Class;
    rdfs:subClassOf nb:Session.

nb:PhenotypicSession a rdfs:Class;
    rdfs:subClassOf nb:Session.

...

nb:Session a rdfs:Class.
```
from https://github.com/neurobagel/recipes/blob/main/vocab/nb_vocab.ttl.

When we first select for all sessions in the Neurobagel query:

https://github.com/neurobagel/api/blob/d2e090b412c365f169f31c028351fc5457a9cfd5/docs/default_neurobagel_query.rq#L19-L20

This will return `ImagingSession`, `PhenotypicSession`, and `Session` for session_type, since an instance of the first two is inferred through RDF inference to also be an instance of `Session`. By default in RDF, each class is a subclass of itself.

### PR DESCRIPTION
**Pull request by alyssadai**
- Fixes #374 
- Closes #361 

Changes proposed in this pull request:

- Update query template to exclude instances of `Session` when matching sessions (returned by default due to RDF inferencing based on class relationships)
- Run integration tests using a test graph as part of CI

## Checklist
_This section is for the PR reviewer_

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see our [Contributing Guidelines](https://neurobagel.org/contributing/CONTRIBUTING#pull-request-guidelines) for more info)_
- [x] PR has a label for the release changelog or `skip-release` (to be applied by maintainers only)
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [x] Tests pass
- [x] Checks pass
- [x] If the PR changes the SPARQL query template, the default Neurobagel query file has also been [regenerated](https://github.com/neurobagel/api?tab=readme-ov-file#the-default-neurobagel-sparql-query)

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -15,8 +15,20 @@ jobs:
...
+      run: docker compose up -d test_graph
+
+    # TODO: Check if this is necessary, since the Python dependencies setup also takes some time
+    - name: Sleep to allow graph setup
+      uses: jakejarvis/wait-action@master
+      with:
+        time: '5s'
+
+    - name: Check running containers
+      run: docker ps
```
Comment (rmanaem) [inline .github/workflows/test.yaml:31]: Is this so we can check whether containers are running in the workflow logs?

**Review by rmanaem — APPROVED** [review]
:cook:

File: .github/workflows/test.yaml (hunk 1 above)
Comment (alyssadai) [reply .github/workflows/test.yaml:31]: Yes, just as a sanity check :)

**Comment by alyssadai** [comment]
Note that our public nodes need to be redeployed with this fix.

**Pull request merged**